fix(docs): 2/5 grammar in `book/` (#2214)
tact-lang/tact: docs/src/content/docs/book/contracts.mdx

### Supported interfaces, `@interface(…)` {#interfaces}

> It's difficult to determine what a contract does and which [receivers](#receiver-functions) and [getters](#getter-functions) it has without examining its source code. Sometimes the source is unavailable or inaccessible, leaving one to attempt disassembling and introspecting the contract, which is a very messy, error-prone approach with diminishing returns and no real reproducibility.

Reviewer: ```suggestion
It's hard to determine what a contract does and which [receivers](#receiver-functions) and [getters](#getter-functions) it has without examining its source code. Sometimes the source is unavailable or inaccessible, leaving one to try to disassemble and introspect the contract, which is a very messy and error-prone approach with diminishing returns and no real reproducibility.
```

> To resolve this issue, an [OTP-001: Supported Interfaces](/ref/evolution/otp-001) was created. According to it, Tact contracts [can report](/book/config#options-interfacesgetter) the list of supported interfaces through a special `supported_interfaces` [getter](#getter-functions). That getter is accessible off-chain using any TON Blockchain explorer—one simply needs to specify `supported_interfaces` as a method to execute and obtain a list of hexadecimal values.

Reviewer: ```suggestion
To resolve this issue, an [OTP-001: Supported Interfaces](/ref/evolution/otp-001) was created. According to it, Tact contracts [can report](/book/config#options-interfacesgetter) the list of supported interfaces through a special `supported_interfaces` [getter](#getter-functions). That getter is accessible off-chain using any TON Blockchain explorer — one simply needs to specify `supported_interfaces` as a method to execute and obtain a list of hexadecimal values.
```

}
```

Tact provides a small set of interfaces under specific conditions:

> * `"org.ton.abi.ipfs.v0"{:tact}`, according to [OTP-003: Self-ABI Reporting](/ref/evolution/otp-003)—opt-in via [`ipfsAbiGetter`](/book/config#options-ipfsabigetter) config property

Reviewer: ```suggestion
* `"org.ton.abi.ipfs.v0"{:tact}`, according to [OTP-003: Self-ABI Reporting](/ref/evolution/otp-003) — opt-in via [`ipfsAbiGetter`](/book/config#options-ipfsabigetter) config property
```

};
```

Additionally, [`tact.config.json`](/book/config) may still be used in [Blueprint][bp] projects. In such cases, values specified in [`tact.config.json`](/book/config) act as defaults unless modified in the `wrappers/`.

:::note

  If you have the `separateCompilables` option set to `true{:typescript}` in the [`blueprint.config.ts`][bp-config], then the `.compile.ts` files will be located in the `compilables/` directory, and **not** in `wrappers/`.

:::

:::caution

>   Be aware that adding an interface does not guarantee that the contract actually implements any particular functionality or that it implements it in any particular way. It is simply an off-chain, verifiable promise indicating that a contract _might_ contain some specific code. It is your responsibility to trust—but verify—such claims.

Reviewer: ```suggestion
  Be aware that adding an interface does not guarantee that the contract actually implements any particular functionality or that it implements it in any particular way. It is simply an off-chain, verifiable promise indicating that a contract _might_ contain some specific code. It is your responsibility to trust, but verify, such claims.
```

  In addition, there is no guarantee that there will not be name clashes between different interfaces, although they are unlikely because even the first 128 bits of SHA-256 provide sufficient [collision resistance](https://en.wikipedia.org/wiki/Collision_resistance).

:::

### Contract parameters {#parameters}

<Badge text="Available since Tact 1.6" variant="tip" size="medium"/><p/>

Most contracts do not require any on-chain deployment logic that runs just once after the contract's deployment is completed. As such, the costly lazy deployment strategy provided by the [`init(){:tact}`](#init-function) function is generally discouraged in favor of simple, direct deployments with [initial code and initial data](/book/expressions#initof).

To declare [persistent state variables](#variables) and set their values during such deployments, use the following _contract parameters_ syntax:

```tact
contract ContractParameters(
    // persistent state variables declared via the contract parameters syntax

>     val: Int,              // Int

Reviewer: ```suggestion
    val: Int,              // Int, implicitly serialized to a 257-bit signed
```

>     val32: Int as uint32,  // Int serialized to a 32-bit unsigned integer

Reviewer: ```suggestion
    val32: Int as uint32,  // Int, serialized to a 32-bit unsigned
```

:::note

  To obtain the initial state of the target contract in [internal functions](#internal-functions), [receivers](#receiver-functions), or [getters](#getter-functions), use the [`initOf{:tact}`](/book/expressions#initof) expression.

  To obtain only the contract's code, use the [`codeOf{:tact}`](/book/expressions#codeof) expression.

:::

### Persistent state variables {#variables}

Contracts can define state variables that persist between contract calls, thus commonly referred to as _storage_ variables. Contracts in TON [pay rent](https://docs.ton.org/develop/smart-contracts/fees#storage-fee) in proportion to the amount of persistent space they consume, so [compact representations via serialization](/book/integers#serialization) are encouraged.

The storage variables can be defined by using [contract parameters](#parameters) or by using the following _contract fields_ syntax:

```tact
contract ContractFields {
    // persistent state variables declared via the contract fields syntax

>     val: Int;              // Int

Reviewer: ```suggestion
    val: Int;              // Int, implicitly serialized to a 257-bit signed
```

>     val32: Int as uint32;  // Int serialized to a 32-bit unsigned

Reviewer: ```suggestion
    val32: Int as uint32;  // Int, serialized to a 32-bit unsigned
```

### Internal functions

> These functions behave similarly to private methods in popular object-oriented languages—they're internal to contracts and can be called by prefixing them with a special [identifier `self{:tact}`](#self). That's why internal functions are sometimes referred to as "contract methods."

Reviewer: ```suggestion
These functions behave similarly to private methods in popular object-oriented languages  —they're internal to contracts and can be called by prefixing them with a special [identifier `self{:tact}`](#self). That's why internal functions are sometimes referred to as "contract methods."
```
